pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -608,14 +608,26 @@ func (c *Reconciler) createTaskRun(ctx context.Context, rprt *resources.Resolved
 		return c.PipelineClientSet.TektonV1beta1().TaskRuns(pr.Namespace).UpdateStatus(tr)
 	}
 
+	// Propagate labels and annotations from PipelineRun and TaskSpec to TaskRun.
+	labels := getTaskrunLabels(pr, rprt.PipelineTask.Name)
+	annotations := getTaskrunAnnotations(pr)
+	if rprt.PipelineTask.TaskSpec != nil {
```

Comment:
One question here : which labels should take precedence ? My gut is that the `TaskRun` labels should take precedence over the one defined in the `Pipeline` ; otherwise a `Pipeline` can define a label that override `tekton.dev/Task` or something and it could break some UX/UI/…. This code does the other way around (labels define in the `Pipeline` tasks take over the `TaskRun` labels).

Reply:
the way its implemented, labels defined in pipeline would take precedence. You are right, it is possible that pipeline defined `tekton.dev` label would break and takes over the TaskRun labels. I will change the order here.

Should we apply the same precedence order in [Conditions](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/pipelinerun/pipelinerun.go#L812)? I can create a separate PR for conditions.

Also, I think labels with `tekton.dev/` should be reserved for the platform and we should not allow such labels, anything starting with `tekton.dev/`. What do you think? We could add this kind of validation in this PR as part of [ValidateTaskSpecMetadata](https://github.com/tektoncd/pipeline/pull/2826/files#diff-06d577762a729d5f494cfec3dd6771a6R157).

Reply:
> Should we apply the same precedence order in [Conditions](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/pipelinerun/pipelinerun.go#L812)? I can create a separate PR for conditions.

I would think so indeed :wink: 

> Also, I think labels with `tekton.dev/` should be reserved for the platform and we should not allow such labels, anything starting with `tekton.dev/`. What do you think? We could add this kind of validation in this PR as part of [ValidateTaskSpecMetadata](https://github.com/tektoncd/pipeline/pull/2826/files#diff-06d577762a729d5f494cfec3dd6771a6R157).

Indeed, we should probably add that kind of validation to `ValidateTaskSpecMetadata` (for some reason I thought that was already the case :stuck_out_tongue_winking_eye: )

Reply:
Nope, in general `Metadata` is validated for valid `Name` (no special character `.` and max length of 63 characters). I have added validation on labels as part of `taskSpec` metadata to restrict using `tekton.dev/Task`. Not sure if we can block in general any label with `tekton.dev` (`git` and `docker` credentials use `tekton.dev` but they belong to k8s `Secrets`) in `taskSpec` metadata or restrict labels with `tekton.dev/` from any Tekton resource i.e. `Pipeline`, `Task`, etc

Reply:
I have updated the order so that TaskRun labels takes higher precedence.